packages/legend-graph/src/graphManager/protocol/pure/v1/V1_PureGraphManager.ts

```diff
@@ -3507,10 +3520,16 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
     const graphData = new V1_PureModelContextData();
     const dependencyManager = graph.dependencyManager;
     const generatedModel = graph.generationModel;
-    graphData.elements = [
-      ...dependencyManager.allOwnElements,
-      ...generatedModel.allOwnElements,
-    ].map((element) => this.elementToProtocol(element));
+    let elements = [...generatedModel.allOwnElements];
+    if (dependencyManager.origin instanceof GraphEntities) {
```

Comment:
Let's add a small comment here to document why we're doing this, the logic looks convoluted enough.